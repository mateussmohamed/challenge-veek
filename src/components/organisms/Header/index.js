import React from 'react'
import styled from 'styled-components'

import { Button, Logo, Link } from 'atoms'
import { Navbar } from 'molecules'

const HeaderStyled = styled.header`
  @media only screen and (min-width: 0em) {
    width: 100%;
    position: absolute;
    top: 0;
    left: 0;
    padding: 25px 0;
  }
`

const HeaderLogo = styled(Link)`
  @media only screen and (min-width: 64em) {
    order: 1;
    margin: 0;
  }
`

const NavLink = styled(Link)`
  color: #000;
  :nth-of-type(n + 5) {
    order: 2;
  }

  @media only screen and (min-width: 0em) {
    font-size: 16px;
  }

  @media only screen and (min-width: 48em) {
    margin: 0;
    padding: 0;
    text-align: normal;
  }
`

const ApplyButton = styled(Button)`
  min-width: 135px;
  order: 9;

  @media only screen and (min-width: 64em) {
    position: absolute;
    right: -150px;
  }
  @media only screen and (min-width: 75em) {
    right: -230px;
  }
`

const Header = () => (
  <HeaderStyled>
    <Navbar>
      <HeaderLogo className="logo">
        <Logo color="#eb5b49" />
      </HeaderLogo>
      <NavLink>Team</NavLink>
      <NavLink>About us</NavLink>
      <NavLink>Process</NavLink>
      <NavLink>Explication</NavLink>
      <NavLink>New</NavLink>
      <NavLink>Contact</NavLink>
      <ApplyButton secondary>Apply</ApplyButton>
    </Navbar>
  </HeaderStyled>
)

export default Header
