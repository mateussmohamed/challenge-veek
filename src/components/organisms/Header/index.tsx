import React from 'react'
import styled from 'styled-components'

import { Button, Logo, Link, NavItem } from '../../atoms'
import { Navbar } from '../../molecules'

const HeaderStyled = styled.header`
  @media only screen and (min-width: 0em) {
    width: 100%;
    position: absolute;
    top: 0;
    left: 0;
    padding: 25px 0;
  }
`

const HeaderLogo = styled(Link)`
  @media only screen and (min-width: 64em) {
    order: 1;
    margin: 0;
  }
`

const ApplyButton = styled(Button)`
  min-width: 135px;
  order: 9;

  @media only screen and (min-width: 64em) {
    position: absolute;
    right: -150px;
  }
  @media only screen and (min-width: 75em) {
    right: -230px;
  }
`

export const Header = () => (
  <HeaderStyled>
    <Navbar>
      <HeaderLogo>
        <Logo color="#eb5b49" />
      </HeaderLogo>
      <NavItem>Team</NavItem>
      <NavItem>About us</NavItem>
      <NavItem>Process</NavItem>
      <NavItem>Explication</NavItem>
      <NavItem>New</NavItem>
      <NavItem>Contact</NavItem>
      <ApplyButton secondary>Apply</ApplyButton>
    </Navbar>
  </HeaderStyled>
)
