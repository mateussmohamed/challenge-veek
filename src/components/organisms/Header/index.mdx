---
name: Header
menu: Organisms
---

import { Playground, PropsTable } from 'docz'
import Header from './index'

# Header
<PropsTable of={Header} />

<Playground>
    <Header />
</Playground>
