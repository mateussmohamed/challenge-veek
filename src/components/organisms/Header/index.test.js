import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import Header from './index'

describe('[<Header />]', () => {
  const wrap = (props = {}) => shallow(<Header {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap()

    expect(wrapper).toMatchSnapshot()
  })
})
