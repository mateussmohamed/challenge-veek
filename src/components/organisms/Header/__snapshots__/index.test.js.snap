// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<Header />] matches to snapshot 1`] = `
<Header__HeaderStyled>
  <Navbar>
    <Header__HeaderLogo
      className="logo"
    >
      <Logo
        className="logo"
        color="#eb5b49"
        height="35px"
        width="90px"
      />
    </Header__HeaderLogo>
    <NavLink__StyledNavLink>
      Team
    </NavLink__StyledNavLink>
    <NavLink__StyledNavLink>
      About us
    </NavLink__StyledNavLink>
    <NavLink__StyledNavLink>
      Process
    </NavLink__StyledNavLink>
    <NavLink__StyledNavLink>
      Explication
    </NavLink__StyledNavLink>
    <NavLink__StyledNavLink>
      New
    </NavLink__StyledNavLink>
    <NavLink__StyledNavLink>
      Contact
    </NavLink__StyledNavLink>
    <Header__ApplyButton
      secondary={true}
    >
      Apply
    </Header__ApplyButton>
  </Navbar>
</Header__HeaderStyled>
`;
